File: backend/src/main/java/com/serhiienko/backend/model/form/AppointmentRequest.java
package com.serhiienko.backend.model.form;

import com.serhiienko.backend.model.entity.Pricing;
import com.serhiienko.backend.model.enumeration.AppointmentStatus;
import com.serhiienko.backend.model.enumeration.EstimatedTime;
import lombok.Data;

@Data
public class AppointmentRequest {
    private AppointmentStatus status;
    private String doctorFullName;
    private String patientEmail;
    private String date;
    private Pricing pricing;
    private boolean paid;
}

File: frontend/src/models/AppointmentRequest.ts
import PricingDTO from "./PricingDTO";

class AppointmentRequest{
    status: string;
    doctorFullName: string;
    pricing?: PricingDTO;
    patientEmail?: string;
    date?: string;
    
    constructor(status: string, doctor: string, pricing?: PricingDTO, patient?: string, date?: string, ){
        this.status = status;
        this.doctorFullName = doctor;
        this.pricing = pricing;
        this.patientEmail = patient;
        this.date = date;
    }
}

export default AppointmentRequest;